### PR TITLE
feat: add csv export functionality for code review table

### PR DESCRIPTION
**Pull request by ganesh-nithin**
# Why this change is required

This PR aims to add csv export functionality for code review table.

# Describe your changes

- add CSV export functionality

# How this change has been tested

Tested UI in browser. CSV export tested in Excel.

# Reference

image
image
image
image

# Checklist

- [x] I have performed a self review of my code and intend to submit as such
- [ ] I have added necessary explanations and inline comments for review
- [ ] I have considered updating necessary README or other documentations
- [ ] I have added/modified necessary automated tests
- [ ] This includes a breaking changes and needs to be listed in release notes

### CODE REVIEW

File: client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloader.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { useCSVDownloader } from "react-papaparse";
+
+import { SENTENCE_JOINER } from "src/constants/constants";
+import { IFetchedCodeReviewPullRequest } from "src/services/api/api";
+import { getFormattedDateWithTime } from "src/utils/utils";
+
+import styles from "./CodeReviewCSVDownloader.module.scss";
+import { CodeReviewCSVDownloaderUtils } from "./CodeReviewCSVDownloaderUtils";
+import { CodeReviewDisplayHoursToDaysUtil } from "../../CodeReviewDisplayHoursToDays/codeReviewDisplayHoursToDaysUtils";
```
Comment (ganesh-nithin) [inline client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloader.tsx:9]: We have moved methods in this class to global utils, this import statement will get updated when #110 is merged to main.

File: client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloader.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+interface ICodeReviewMetricsTableProps {
+  pullRequests: IFetchedCodeReviewPullRequest[];
+}
+
+export const CSVDownloader = ({ pullRequests }: ICodeReviewMetricsTableProps) => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const { CSVDownloader } = useCSVDownloader();
+
+  const buttonType = "button";
+  const fileName = "code-review-metrics";
```
Comment (sathish-thangavel) [inline client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloader.tsx:19]: Can we move these constants outside the react component ?

File: client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloader.module.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+@import "src/styles/index.scss";
+
+.downloadButton {
+  background-color: $app-quaternary-background-color;
+  border: 0;
+  border-radius: $spacing-xs;
```
Comment (sathish-thangavel) [inline client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloader.module.scss:6]: Use border-radius variable for this

File: client/src/constants/common.constants.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { DateTime } from "luxon";
 
 export const NOT_AVAILABLE = "-";
+export const SENTENCE_JOINER = ", ";
```
Comment (sathish-thangavel) [inline client/src/constants/common.constants.ts:4]: Can we rename to CSV_SENTENCE_JOINER ?

File: client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloaderUtils.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  static getFormattedVotesTimeline(votesTimeline: IFetchedPullRequestVotesTimeline[]) {
+    if (!votesTimeline) {
+      return NOT_AVAILABLE;
+    }
+
+    return votesTimeline
+      .map((vote) => `${vote.author} (${vote.vote} - ${getFormattedDateWithTime(vote.timeOfVote)})`)
+      .join(SENTENCE_JOINER);
```
Comment (sathish-thangavel) [inline client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloaderUtils.ts:21]: If this variable is used only here . Can we declare it as private variable inside this ?

**Review by sathish-thangavel — CHANGES_REQUESTED** [review]

File: client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloaderUtils.ts (hunk 5 above)
Comment (ganesh-nithin) [reply client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloaderUtils.ts:21]: It has been used in component too, so I made declared it in common constanmts.

Comment (ganesh-nithin) [reply client/src/components/containers/CodeReviewMetricsContainers/CodeReviewMetricsTable/CodeReviewCSVDownloader/CodeReviewCSVDownloaderUtils.ts:21]: This is moved to component constant file.

**Review by sathish-thangavel — APPROVED** [review]

**Pull request merged**